coreneuron/utils/progressbar/progressbar.c

```diff
@@ -94,7 +98,13 @@ void progressbar_free(progressbar* bar) {
 void progressbar_update(progressbar* bar, unsigned long value, double t) {
     bar->value = value;
     bar->t = t;
-    progressbar_draw(bar);
+    // The progress bar will be printed only if the minimum interval
+    // to print max BAR_UPDATES_NUMBER bars has passed from the
+    // previous progress bar printing.
+    if ((bar->value - bar->prev_value) >= bar->max / BAR_UPDATES_NUMBER) {
```

Comment:
`max->max` and `BAR_UPDATES_NUMBER` are both integral types, division will be 0 if `BAR_UPDATES_NUMBER > bar->max`, so it's gonna always print.
Some casts are needed

Reply:
You are right, however if bar->max, which is the number of simulation steps, is smaller that the BAR_UPDATE_NUMBER there are still going to be less than BAR_UPDATE_NUMBER of bars printed, which is the intended behavior.